src/java.base/macosx/classes/sun/nio/fs/MacOSXFileSystem.java

```diff
@@ -35,6 +36,17 @@
 
 class MacOSXFileSystem extends BsdFileSystem {
 
+    private static final String PROPERTY_ENABLE_FILE_NAME_ENCODING =
+        "jdk.nio.path.useNormalizationFormD";
+
+    private static final boolean ENCODE_FILE_NAMES;
+
+    static {
+        final String name = PROPERTY_ENABLE_FILE_NAME_ENCODING;
+        String value = GetPropertyAction.privilegedGetProperty(name);
```

Comment:
Can we use `sun.security.action.GetBooleanAction#privilegedGetProperty` instead?

Reply:
> Can we use `sun.security.action.GetBooleanAction#privilegedGetProperty` instead?

That is problematic for the case that a value is empty, as in `-Djdk.nio.path.useNormalizationFormD`.

Reply:
I'm not sure I understand. What the problem with empty value exactly?

Reply:
There are 4 cases:
jdk.nio.path.useNormalizationFormD not set
-Djdk.nio.path.useNormalizationFormD
-Djdk.nio.path.useNormalizationFormD=true or a case variant
-Djdk.nio.path.useNormalizationFormD=false or a case variant
The second and third should enable the normalisation. This will do it
```
    NORMALIZE_FILE_PATHS = (value != null)
                && ("".equals(value) || Boolean.parseBoolean(value));
```

Reply:
>The second and third should enable the normalisation.
Is it convention for all JDK system properties?
Then I don't understand why `privilegedGetProperty` doesn't have exactly the same logic as you propose.
Perhaps another helper method could be introduced?